include/ensmallen_bits/moead/moead.hpp

```diff
@@ -18,35 +16,38 @@
 #ifndef ENSMALLEN_MOEAD_MOEAD_HPP
 #define ENSMALLEN_MOEAD_MOEAD_HPP
 
+//! Decomposition policies
+#include "decomposition_policies/tchebycheff_decomposition.hpp"
+#include "decomposition_policies/weighted_decomposition.hpp"
+#include "decomposition_policies/pbi_decomposition.hpp"
+
+//! Weight Init policies
+#include "weight_init_policies/bbs_init.hpp"
 namespace ens {
 
 /**
- * This class implements the MOEA/D algorithm with Differential Evolution
- * crossover. Step numbers used in different parts of the implementation
- * correspond to the step number used in the original algorithm by the author.
+ * MOEA/D-DE (Multi Objective Evolutionary Algorithm based on Decompositon - 
+ * Differential Variant) is a multiobjective optimization algorithm. This class 
+ * implements the said optimizer. 
+ *
+ * The algorithm works by generating a candidate population from a fixed starting point. 
+ * Reference directions are generated to guide the optimization process towards the Pareto Front. 
+ * Further, a decomposition function is defined to decompose the problem to a scalar optimization 
+ * objective. Utilizing genetic operators, offsprings are generated with better decomposition values 
+ * to replace the neighboring parent solutions. 
  *
  * For more information, see the following:
  * @code
- * @article{article,
- * author = {Zhang, Qingfu and Li, Hui},
- * year = {2008},
- * pages = {712 - 731},
- * title = {MOEA/D: A Multiobjective Evolutionary Algorithm Based on
- *    Decomposition},
- * journal = {Evolutionary Computation, IEEE Transactions on},
- *
- * @article{4633340,
- * author={H. {Li} and Q. {Zhang}},
- * year={2009},
- * pages={284-302},}
- * title={Multiobjective Optimization Problems With Complicated Pareto Sets, MOEA/D and NSGA-II},
- * journal={IEEE Transactions on Evolutionary Computation},
+ * @article{li2008multiobjective,
+ * title={Multiobjective optimization problems with complicated Pareto sets, MOEA/D and NSGA-II},
+ * author={Li, Hui and Zhang, Qingfu},
+ * journal={IEEE transactions on evolutionary computation},
+ * pages={284--302},
+ * year={2008},
  * @endcode
- *
- * MOEA/D can optimize arbitrary multi-objective functions. For more details,
- * see the documentation on function types included with this distribution or
- * on the ensmallen website.
  */
+template<typename InitPolicyType = BayesianBootstrap,
+         typename DecompPolicyType = Tchebycheff>
```

Comment:
I think it would be useful to call this `MOEADType` now and provide some aliases, see https://github.com/mlpack/ensmallen/blob/3497f276f4f8c399bcc565f723735a8dd7c86864/include/ensmallen_bits/sgd/sgd.hpp#L247-L253 for an example, that way we can keep using `MOEAD` in the tests and I think users will most probably go for the default policies as well, and in this case it seems odd to use `MOEAD<>`. Let me know what you think.

Reply:
Okay, but we would need to write 3 (decomposition) x 4 (init) = 12 aliases.

Reply:
And how do we name each of them, any ideas?

Reply:
I would just go with `MOEA` for now, a user can always use `MOEAType<...>` or whatever we call this, and specify the policies same as the user would do right now.

Reply:
How about

```
using DefaultMOEAD = MOEAD<BayesianBootStrap, Tchebycheff>;
```

Reply:
I think that works.